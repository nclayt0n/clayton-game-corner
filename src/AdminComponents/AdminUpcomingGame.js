import React from 'react';
import {withRouter} from 'react-router-dom';
import GameApiService from '../services/game-api-services';
import config from '../config';
import Context from '../Context';
import ValidationError from '../Validation/ValidationError';

class AdminUpcomingGame extends React.Component{
    static contextType=Context;
    constructor(props){
        super()
        this.state={
            error:'',
            game:props.game
        };
    }
    
    onDelete(id,context){
        GameApiService.deleteGame(`${config.API_ENDPOINT}/api/game/upcoming/${id}`)
            .then(context.deleteUpcomingGame(id))
                .catch(error =>{
                 this.setState({error:error.message});
                });
            this.props.history.push('/game/upcoming');
    }
    onSubmit(e,id,context){
        e.preventDefault();
        let date;
        (e.target.newGameDate.value.length===0)?date=this.props.game.date:date=e.target.newGameDate.value;
        let updatedGame={
            id,
            title:e.target.title.value,
            game_type:e.target.game_type.value,
            date
        };
        GameApiService.patchUpcoming(`${config.API_ENDPOINT}/api/game/upcoming/${id}`,updatedGame)
            .then(context.updateUpcomingGame(updatedGame))
                .catch(error =>{
                    this.setState({error:error.message});
                });
    }
    render(){
        return(
       <section key={this.props.game.id}>
                <form key={this.props.game.id} onSubmit={(e)=>this.onSubmit(e,this.props.game.id,this.context)}>
                    <label htmlFor='gameDate'> Current Date: {this.state.game.date}
                    </label>
                    <label htmlFor='newGameDate'>New Date:
                        <input type="date" name='newGameDate'/>
                    </label>
                    <label htmlFor='title'>Title:
                        <input type="text" name='title' defaultValue={this.props.game.title}/>
                    </label>
                    <label htmlFor='game_type'> Game Type: 
                        <select name='game_type'>
                            <option value={this.props.game.game_type}>{this.props.game.game_type}</option>
                            <option value={this.props.game.game_type==='video'?'tabletop':'video'}>{this.props.game.game_type==='video'?'tabletop':'video'}</option>
                        </select>
                    </label>
                    <button type='submit'>Update</button>
                </form>
                
                <button type='button' onClick={()=>this.onDelete(this.props.game.id,this.context)}>Delete</button>
                <ValidationError errorMessage={this.state.error}/>
            </section>)
    }
}
export default withRouter(AdminUpcomingGame);