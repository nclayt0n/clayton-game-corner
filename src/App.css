/* http://meyerweb.com/eric/tools/css/reset/ 
   v2.0 | 20110126
   License: none (public domain)
*/

html,
body,
div,
span,
applet,
object,
iframe,
h1,
h2,
h3,
h4,
h5,
h6,
p,
blockquote,
pre,
a,
abbr,
acronym,
address,
big,
cite,
code,
del,
dfn,
em,
img,
ins,
kbd,
q,
s,
samp,
small,
strike,
strong,
sub,
sup,
tt,
var,
b,
u,
i,
center,
dl,
dt,
dd,
ol,
ul,
li,
fieldset,
form,
label,
legend,
table,
caption,
tbody,
tfoot,
thead,
tr,
th,
td,
article,
aside,
canvas,
details,
embed,
figure,
figcaption,
footer,
header,
hgroup,
menu,
nav,
output,
ruby,
section,
summary,
time,
mark,
audio,
video {
    margin: 0;
    padding: 0;
    border: 0;
    font-size: 100%;
    font: inherit;
    vertical-align: baseline;
}


/* HTML5 display-role reset for older browsers */

article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
menu,
nav,
section {
    display: block;
}

body {
    line-height: 1;
}

ol,
ul {
    list-style: none;
}

blockquote,
q {
    quotes: none;
}

blockquote:before,
blockquote:after,
q:before,
q:after {
    content: '';
    content: none;
}

table {
    border-collapse: collapse;
    border-spacing: 0;
}

:root {
    --mainColor: rgb(17, 72, 82);
}

img {
    width: 200px;
    height: 200px;
}

main {
    margin-bottom: 150px;
}

nav {
    background-color: white;
}

article,
section {
    display: flex;
    flex-direction: column;
    text-align: center;
    border: 1px solid black;
    width: 300px;
    margin: 80px auto;
    padding: 15px;
}

p {
    padding: 10px;
    margin: auto;
}

textarea {
    border: 1px solid black;
}

footer {
    position: fixed;
    bottom: 0;
    display: flex;
    width: 100%;
    background-color: white;
}

h2,
h1 {
    font-size: 22px;
    text-align: center;
}

h3 {
    margin: 20px auto;
    text-align: center;
}

footer address {
    margin: auto;
}

form {
    display: flex;
    flex-direction: column;
    flex: 1;
}

form textarea {
    height: 150px;
}

button {
    border: 1px solid black;
    margin: auto;
    width: 150px;
    background-color: var(--mainColor);
    color: white;
}

header {
    position: fixed;
    top: 0;
    width: 100%;
    text-align: center;
    width: 100%;
    background-color: var(--mainColor);
}

aside {
    margin: 120px auto 20px;
}

#headerLogo {
    height: 90px;
    width: 80px;
    margin: 10px;
}

#loginContainer {
    margin-top: 150px;
}

#loginForm {
    width: 300px;
}

#loginForm input {
    width: 200px;
}

.notFoundPage {
    margin-top: 150px;
}