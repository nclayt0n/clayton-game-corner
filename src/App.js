import React from 'react';
import {Switch,Route} from 'react-router-dom';
import Context from './Context';
import LandingPage from './Components/LandingPage';
import PublicOnlyRoute from './Utils/PublicOnlyRoute';
import PrivateRoute from './Utils/PrivateRoute';
import TabletopGameReview from './Components/TabletopGameReviews';
import VideoGameReview from './Components/VideoGameReview';
import UpcomingGameList from './Components/UpcomingGameList';
import './App.css';
import NotFound from './Components/NotFound';
import Header from './Components/Header';
import Nav from './Components/Nav';
import Footer from './Components/Footer';
import UpcomingGameInfo from './Components/UpcomingGameInfo';
import AdminLandingPage from './AdminComponents/AdminLandingPage';
import AdminReview from './AdminComponents/AdminReviewList';
import AdminReviewList from './AdminComponents/AdminReviewList';
import AdminLogin from './AdminComponents/AdminLogin';
import AdminUpcomingGameList from './AdminComponents/AdminUpcomingGameList';
import TokenService from './services/token-service'

class App extends React.Component {
  constructor(){
    super()
      this.state={
          user_id:0,
          reviews:[],
          bio:'',
          upcomingGames:[]
      };
    }
    handleAddReviews=(reviews)=>{
    this.setState({
      reviews:[...reviews]
    })
  }
  handleAddUpcomings=(upcomingGames)=>{
    this.setState({
      upcomingGames:[...upcomingGames]
    })
  }
  handleAddBio=(bio)=>{
    this.setState({
      bio:bio
    })
  }
  render(){
    let userId;
    (TokenService.getAuthToken()===null)?userId=0
      :userId=TokenService.decodeAuthToken(TokenService.getAuthToken());
    
    const contextValue={
      user_id:userId,
      bio:this.state.bio,
      upcomingGames:this.state.upcomingGames,
      reviews:this.state.reviews,
      addReviews:this.handleAddReviews,
      addUpcomingGames:this.handleAddUpcomings,
      addBio:this.handleAddBio
    };
  return (
    <main className='App'>
        <Context.Provider value={contextValue}>
        <Switch>
          <Route exact path='/' component={LandingPage}/>
          <Route path='/game/review/tabletop' component={TabletopGameReview}/>
          <Route path='/game/review/video' component={VideoGameReview}/>
          <Route exact path='/game/upcoming' component={UpcomingGameList}/>
          <Route exact path='/game/upcoming/info' component={UpcomingGameInfo}/>
          <Route path='/nav' component={Nav}/>
          <PrivateRoute exact path='/admin' component={AdminLandingPage}/>
          <PrivateRoute exact path='/admin/game/review-list' component={AdminReviewList}/>
          <PrivateRoute exact path='/admin/game/upcoming-list' component={AdminUpcomingGameList}/>
          <Route exact path='/admin/login' component={AdminLogin}/>
          <Route component={NotFound}/>
        </Switch>
        </Context.Provider>
      <Footer/>
    </main>
  );
  }
}

export default App;