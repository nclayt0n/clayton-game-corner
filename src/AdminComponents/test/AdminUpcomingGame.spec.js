import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter,StaticRouter} from 'react-router-dom';
import App from '../../App';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AdminUpcomingGame from '../AdminUpcomingGame';

describe('AdminUpcomingGameList',()=>{
    let games=[{
            "title": "Upcoming test Game 1",
            "date": "02/01/2020",
            "game_type": "video"
        },
        {
            "title": "Upcoming test Game 2",
            "date": "03/01/2020",
            "game_type": "tabletop"

        }];
  it('renders without crashing', () => {
     
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <App>
                        {games.map(game=>{
                            return <AdminUpcomingGame key={39} game={game}/>
                        })}
                    </App>
                </BrowserRouter>,
                div
              );
            ReactDOM.unmountComponentAtNode(div);
        });
    it('snapshot', () => {
        const wrapper=shallow(
            <StaticRouter>
                {games.map(game=>{
                    return <AdminUpcomingGame key={42} game={game}/>
                })} 
            </StaticRouter>)
            expect(toJson(wrapper))
            .toMatchSnapshot()
    });
})