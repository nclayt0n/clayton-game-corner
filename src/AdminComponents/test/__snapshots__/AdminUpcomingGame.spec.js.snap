// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminUpcomingGameList snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  staticContext={Object {}}
>
  <withRouter(AdminUpcomingGame)
    game={
      Object {
        "date": "02/01/2020",
        "game_type": "video",
        "title": "Upcoming test Game 1",
      }
    }
    key="42"
  />
  <withRouter(AdminUpcomingGame)
    game={
      Object {
        "date": "03/01/2020",
        "game_type": "tabletop",
        "title": "Upcoming test Game 2",
      }
    }
    key="42"
  />
</Router>
`;
